syntax = "proto3";

package android.jsonpb;

// Note: this file explicitly uses names that does NOT follow the Protobuf Style Guide for
// testing purposes. When writing a .proto file as a JSON schema, you should:
// - Follow the Protobuf Style Guide for field names / enum value names
// - If the JSON file is going to have field names that does not conform to the Protobuf
//   Style Guide, use json_name option to indicate an alternative name.

message NoJsonName {
    string foo_bar = 1;
    string barBaz = 2;
    string BazQux = 3;
    string QUX_QUUX = 4;
}

message WithJsonName {
    string foo_bar = 1 [json_name = "FOOBAR"];
    string barBaz = 2 [json_name = "BarBaz"];
    string BazQux = 3 [json_name = "baz_qux"];
    string QUX_QUUX = 4 [json_name = "quxQuux"];
}

message Parent {
    NoJsonName no_json_name = 1;
    repeated NoJsonName repeated_no_json_name = 2;
    WithJsonName with_json_name = 3;
    repeated WithJsonName repeated_with_json_name = 4;
}
